### PR TITLE
Fix restore VM with allocated root disk

### PR DESCRIPTION
**Pull request by vishesh92**
### Description

This PR fixes the following issues

1. Try to restore a VM twice in stopped state, it will throw an error. 
image

2. Fix the default template selection in reinstallVM modal.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by vishesh92** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
code looks good

@vishesh92 
Can you add 1-2 lines to one of the existing integration tests to cover this test case (restore vm twice)?

File: server/src/main/java/com/cloud/vm/UserVmManagerImpl.java [hunk 1]
```diff
@@ -7909,7 +7909,7 @@ public Pair<UserVmVO, Volume> doInTransaction(final TransactionStatus status) th
 
                 // Detach, destroy and create the usage event for the old root volume.
                 _volsDao.detachVolume(root.getId());
-                _volumeService.destroyVolume(root.getId(), caller, expunge, false);
+                _volumeService.destroyVolume(root.getId(), caller, root.getState() == Volume.State.Allocated || expunge, false);
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/vm/UserVmManagerImpl.java:7912]:
```suggestion
                _volumeService.destroyVolume(root.getId(), caller, Volume.State.Allocated.equals(root.getState()) || expunge, false);
```

**Comment by weizhouapache** [comment]
@blueorangutan package

**Comment by vishesh92** [comment]
@blueorangutan package

**Review by borisstoyanov — APPROVED** [review]
LGTM, manually check both the fixes

**Comment by borisstoyanov** [comment]
Thanks @vishesh92, let's wait the smoketests to come out and merge this.

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by sureshanaparti** [comment]
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test rocky8 kvm-rocky8

**Pull request merged**